frontend/src/components/Communications.vue

```diff
@@ -363,6 +366,24 @@ const { data: membersData } = useQuery({
   query: () => getAllMembers(),
 });
 
+// Fetch current entity (company or speaker) based on props
+const { data: entityData } = useQuery({
+  key: () => [`entity`, props.entityType, props.entityId],
+  query: () => {
+    if (!props.entityId) return Promise.resolve({ data: null } as any);
+    return props.entityType === "company"
+      ? getCompanyById(props.entityId)
+      : getSpeakerById(props.entityId);
+  },
+});
```

Comment:
Is there a need to fetch data? You can pass the entity as a prop as its already available in the parent.